ed25519/point.go

```diff
@@ -43,13 +43,16 @@ func (p *Point) Unmarshal(buf []byte, rem int) ([]byte, int, error) {
 		return buf, rem, surge.ErrUnexpectedEndOfBuffer
 	}
 
-	(*p).SetB32(buf[:PointSizeMarshalled])
+	err := p.SetB32(buf[:PointSizeMarshalled])
+	if err != nil {
+		panic(fmt.Sprintf("can not unmarshal ed25519 point : %v", err))
```

Comment:
This shouldn't be a panic: this would allow a malicious actor to send us bad data and cause us to crash. Instead, the error should be returned.